### ISSUE TITLE
mtls fails with 503 in run long running test after nodeagent restart

### ISSUE THREAD

**Issue by quanjielin** [istio/istio#14853]
**Problem**
mtls fails with 503 in run long running test after nodeagent restart.

**Repro**
1. using latest master build, trigger long running test
2. restart nodeagent
3. run for a while, mtls fails with 503

**Root cause**
1. nodeagent recreate uds file when restart
2. new uds file fails to mount to istio sidecar container, which cause sds client side fail :
```
[2019-06-12 18:25:06.179][15][trace][connection] [external/envoy/source/common/network/connection_impl.cc:392] [C1] writing 200 bytes, end_stream false
[2019-06-12 18:25:06.179][15][trace][http] [external/envoy/source/common/http/conn_manager_impl.cc:1467] [C1][S3459697968779239420] encoding data via codec (size=535 end_stream=true)
[2019-06-12 18:25:06.179][15][trace][connection] [external/envoy/source/common/network/connection_impl.cc:392] [C1] writing 547 bytes, end_stream false
[2019-06-12 18:25:06.179][15][trace][main] [external/envoy/source/common/event/dispatcher_impl.cc:144] item added to deferred deletion list (size=1)
[2019-06-12 18:25:06.179][15][trace][http] [external/envoy/source/common/http/conn_manager_impl.cc:833] [C1][S3459697968779239420] decode headers called: filter=0x3018aa0 status=1
[2019-06-12 18:25:06.179][15][trace][http] [external/envoy/source/common/http/http1/codec_impl.cc:387] [C1] parsed 105 bytes
[2019-06-12 18:25:06.179][15][trace][connection] [external/envoy/source/common/network/connection_impl.cc:456] [C1] socket event: 2
[2019-06-12 18:25:06.179][15][trace][connection] [external/envoy/source/common/network/connection_impl.cc:541] [C1] write ready
[2019-06-12 18:25:06.179][15][trace][connection] [external/envoy/source/common/network/raw_buffer_socket.cc:66] [C1] write returns: 747
[2019-06-12 18:25:06.179][15][trace][main] [external/envoy/source/common/event/dispatcher_impl.cc:64] clearing deferred deletion list (size=1)
[2019-06-12 18:25:08.177][15][trace][connection] [external/envoy/source/common/network/connection_impl.cc:456] [C1] socket event: 3
[2019-06-12 18:25:08.177][15][trace][connection] [external/envoy/source/common/network/connection_impl.cc:541] [C1] write ready
[2019-06-12 18:25:08.177][15][trace][connection] [external/envoy/source/common/network/connection_impl.cc:494] [C1] read ready
[2019-06-12 18:25:08.177][15][trace][connection] [external/envoy/source/common/network/raw_buffer_socket.cc:23] [C1] read returns: 110
[2019-06-12 18:25:08.177][15][trace][connection] [external/envoy/source/common/network/raw_buffer_socket.cc:37] [C1] read error: Resource temporarily unavailable
[2019-06-12 18:25:08.177][15][trace][http] [external/envoy/source/common/http/http1/codec_impl.cc:366] [C1] parsing 110 bytes
[2019-06-12 18:25:08.177][15][trace][http] [external/envoy/source/common/http/http1/codec_impl.cc:479] [C1] message begin
```

**Comment by duderino** [comment]
@quanjielin does the same thing happen with 1.1.8?

**Comment by duderino** [comment]
@quanjielin told me verbally that this is not a regression so it's not a release blocker.  Still high priority though so we'll get the fix into 1.2.1 and 1.1.10.

**Comment by vaizmanai** [comment]
> @quanjielin told me verbally that this is not a regression so it's not a release blocker. Still high priority though so we'll get the fix into 1.2.1 and 1.1.10.

But it's breaking communication at all, how is not blocker?

**Comment by quanjielin** [comment]
@vaizmanai the fix was merged to 1.2 branch (https://github.com/istio/istio/pull/14883)

**Comment by quanjielin** [comment]
this is done.
